File: 69-sqrtx/sqrtx.java
class Solution {
    public int mySqrt(int x) {
        int r = x;
        int res = x;
        while(res>0){
            double rx = (0.5)*(r+x/r);
            if(rx>r){
                res = (int) (rx-r);
            }else {
                res = (int)(r-rx);
            }
            r = (int)rx;
        }
        if(r<0){
            r = -r;
        }
        return r;
    }
}

File: 69-sqrtx/sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x == 0 || x ==1 ){
            return x;
        }
        long  low,mid,high;
        low = 1;
        high = x;
        int res;
        while (low<=high){
            mid = (low+high)/2;
            if (mid*mid == x){
                return mid;
            }else if(mid*mid>x){
                high = mid -1;
            }else {
                res = mid;
                low = mid +1;
            }
        }
        return res;
    }
};